pyne/vta.py

```diff
@@ -0,0 +1,551 @@
+       bounds[1][0] <= point.y <= bounds[1][-1] and \
+       bounds[2][0] <= point.z <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bound, cor, v_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bound : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    v_1d : float
+        Vector attribute in the given direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    if v_1d > 0:
+        for i in range(len(bound)-1):
+            if bound[i] <= cor < bound[i+1]:
+                idx = i
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if bound[i] < cor <= bound[i+1]:
+                idx = i
+                break
+    # idx could be -1, that means point outside the mesh
+    return idx
+
+def _cal_dir_v(start, end):
+    """
+    Calculate the direction from start to end (start -> end).
+    
+    Parameters:
+    -----------
+    start : Point
+        Start point.
+    end :
+        End point.
+
+    Return:
+    -------
+    v : list
+        direction from start to end, an unit vector.
+    """
+    v_x, v_y, v_z = 0.0, 0.0, 0.0
+    d_x = end.x - start.x
+    d_y = end.y - start.y
+    d_z = end.z - start.z
+    dist = math.sqrt(d_x * d_x + d_y * d_y + d_z * d_z)
+    v_x = d_x / dist
+    v_y = d_y / dist
+    v_z = d_z / dist
+    return [v_x, v_y, v_z]
+
+def _find_next_grid_1d(cor, v_1d, bound):
+    """
+    Find the next grid coordinate of a direction.
+
+    Parameters:
+    -----------
+    cor : float
+        Coordinate of current point in the direction.
+    v_1d : float
+        Direction value.
+    bound : list of float
+        Boundary values of current direction.
+
+    Return:
+    -------
+    n_grid : float
+        The coordinate value of next mesh grid.
+    """
+    n_grid = 0.0
+    flag = False
+    if v_1d > 0:
+        for i in range(len(bound)):
+            if cor < bound[i]:
+                n_grid = bound[i]
+                flag = True
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if cor > bound[i]:
+                n_grid = bound[i]
+                flag = True
+                break
+    if flag:
+        return n_grid
+    else:
+        # point goes out of the mesh
+        return v_1d * float('inf')
+
+
+def _cal_max_travel_length_in_current_voxel(tp, end, v, bounds):
+    """
+    Calculate the maximum travel length in current voxel before reach next
+    boundary.
+    
+    Parameters:
+    -----------
+    tp : Point
+        Temporary point position.
+    end : Point
+        End point
+    v : list
+        Direction vector from start to end: [v_x, v_y, v_z].
+    bounds : list
```

Comment:
Again, I think this could be a little more descriptive in terms of what is expected to be in this set of nested lists.